pkg/kubecost/asset.go

```diff
@@ -2453,13 +2453,13 @@ func (as *AssetSet) AggregateBy(aggregateBy []string, opts *AssetAggregationOpti
 		opts = &AssetAggregationOptions{}
 	}
 
-	if as.IsEmpty() && len(opts.SharedHourlyCosts) == 0 {
```

Comment:
This acquired the read lock to call `IsEmpty()`, relinquished the lock, then immediately acquired the write lock. In this case, contention can leak between these checks, and because go doesn't have upgradeable RW locks, it's better to do any cheap reads inline while holding the write lock.